From 9d0ed3f2d471ec41ff446a32af8e5cb7d7324bdd Mon Sep 17 00:00:00 2001
From: Huibin Hong <huibin.hong@rock-chips.com>
Date: Fri, 1 Mar 2019 15:32:23 +0800
Subject: [PATCH] fiq debugger: ttyS2 and fiq debugger don't use the same uart2

Someone may enable uart2 and fiq debugger, which is illegal.

&uart2 {
	status = "okay";
};

&fiq_debugger {
	status = "okay";
};

Change-Id: Ibecd3c3ca69de1216103d7c373a4d282cf93cb30
Signed-off-by: Huibin Hong <huibin.hong@rock-chips.com>
---
 drivers/soc/rockchip/rk_fiq_debugger.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/soc/rockchip/rk_fiq_debugger.c b/drivers/soc/rockchip/rk_fiq_debugger.c
index 629237da6a6b..00b71e52bc6b 100755
--- a/drivers/soc/rockchip/rk_fiq_debugger.c
+++ b/drivers/soc/rockchip/rk_fiq_debugger.c
@@ -710,6 +710,11 @@ static int __init rk_fiqdbg_probe(struct platform_device *pdev)
 	if (!ok)
 		return -EINVAL;
 
+	if (of_device_is_available(np)) {
+		pr_err("uart%d is enabled, please disable it\n", serial_id);
+		return -EINVAL;
+	}
+
 	/* parse serial hw irq */
 	if (!of_irq_parse_one(np, 0, &oirq))
 		serial_hwirq = oirq.args[1] + 32;
-- 
2.35.3

